deepchecks/datasets/regression/avocado.py

```diff
@@ -0,0 +1,150 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""The avocado dataset contains historical data on avocado prices and sales volume in multiple US markets."""
+import typing as t
+import pandas as pd
+import joblib
+from urllib.request import urlopen
+from deepchecks import Dataset
+
+__all__ = ['load_data', 'load_fitted_model']
+
+_MODEL_URL = 'https://figshare.com/ndownloader/files/32393723'
+_FULL_DATA_URL = 'https://figshare.com/ndownloader/files/32393729'
+_TRAIN_DATA_URL = 'https://figshare.com/ndownloader/files/32393732'
+_TEST_DATA_URL = 'https://figshare.com/ndownloader/files/32393726'
+_target = 'AveragePrice'
+_CAT_FEATURES = ['region', 'type']
+
+
+def load_data(data_format: str = 'Dataset', as_train_test: bool = True) -> \
```

Comment:
a. as I wrote in the slack, why do we need all those options?
b. I'd prefer to have boolean instead of data_format (like sklearn, `as_frame`)

Reply:
About a. - we want to let the user explore the package but also give him the opportunity to experiment a bit with the checks logic. He can do that by manipulating existing datasets and compare the suite results, for example. + It really isn't so hard to implement it...

About b. - This might be an option. But don't you think that it's not so clear what will be the return type if I specify False? I also think that in sklearn it's not clear... that's why I prefer they mention it explicitly

Reply:
a. exactly for the things you wrote he needs train + test datasets. no reason to give him dataframe or single dataset